### PR TITLE
8732-apr-details-pagination

### PR DESCRIPTION
**Pull request by ttoomey**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description

Refactors HUD APR and apr-family reports (CAPER, CeAPR, DQ) to use the drilldown server-side pagination, export, and search from the SPM. We use pagination and async processing to address performance issues with large record sets, such as state-wide reports. Common behaviors between the SPM and APR are extracted to shared concerns. 

- Adds base concern `HudReports::CellDrilldownConcern` and `DrilldownContext` to controller behaviors 
- Adds `HudReports::Cells::SearchQueriesBehavior` concern for search within cell drilldowns
-  Adds `CellDetailExportBase` and `CellDetailExportBuilderBase` to consolidate export logic across all report types
- **APR Family Controllers**: Refactors `HudApr::CellsController` and adds search query controllers for APR, CAPER, CeAPR, and DQ reports
- **SPM Controllers**: Updates `HudSpmReport::CellsController` to use new shared infrastructure
- **Routes**: Adds reusable route concerns in `lib/hud_reports/route_concerns.rb` to DRY up routing patterns
- **View Cleanup**: Removes duplicate XLSX view templates in favor of shared export builders
- **Model Enhancements**: Adds search and scope methods to `AprClient`, removes validation methods from `ReportInstance` (moved to `DrilldownContext`)

### Type of Change
Refactor | New feature

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] I have updated the documentation (or not applicable)
- [x] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: app/controllers/concerns/hud_reports/cells/search_queries_behavior.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      extend ActiveSupport::Concern
+
+      included do
+        # Subclasses must implement:
+        # - report_param_name
+        # - set_report
+        # - generator (must be available via method or before_action)
+        # - build_search_path(query_id)
+        # - build_cell_path
+      end
```
Comment (eanders) [inline app/controllers/concerns/hud_reports/cells/search_queries_behavior.rb]: Is there a reason this is wrapped in the `included do` block?  Does having that block mean something?

File: app/models/hud_reports/cell_detail_export_builder_base.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          final_headers = normalized_headers(headers)
+          sheet.add_row(final_headers.values, style: title)
+
+          # Preload project dependencies before loading clients
+          preload_policies(clients)
+
+          # Use find_each to avoid loading all clients into memory
+          clients.find_each do |client|
+            pii_policy = user.reporting_policy_for_project(
+              project_id: client.project_id,
```
Comment (eanders) [inline app/models/hud_reports/cell_detail_export_builder_base.rb]: might not be worth it, but I wonder if this should be:
```suggestion
              project_id: project_id(client),
```

With:
```
def project_id(record)
  # Override as appropriate in sub classes
  record.project_id
end
```

File: drivers/hud_apr/app/controllers/hud_apr/apr/cells_controller.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ###
-# Copyright 2016 - 2025 Green River Data Analysis, LLC
+# Copyright 2016 - 2026 Green River Data Analysis, LLC
```
Comment (eanders) [inline drivers/hud_apr/app/controllers/hud_apr/apr/cells_controller.rb]: Oh, this is a good reminder we need to queue up the work to update all of them.  Might be best not to update them individually as that tends to generate duplicates.

https://github.com/open-path/Green-River/issues/8738

File: drivers/hud_apr/app/views/hud_apr/shared/_list.haml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                  = client[k]
+                %a{href: '#', class: ['btn', 'btn-sm'], data: { 'bs-toggle' => :popover, title: 'Client Information', url: api_client_path(client.client_id), html: 'true' }}
+                  %i.icon-info
+              - elsif k.to_s == 'destination_client_id'
+                = link_to appropriate_client_path(client[k]) do
+                  = client[k]
+                %a{href: '#', class: ['btn', 'btn-sm'], data: { 'bs-toggle' => :popover, title: 'Client Information', url: api_client_path(client.client_id), html: 'true' }}
+                  %i.icon-info
+              - elsif k.to_s.starts_with?('hiv_aids') && !can_view_hiv_status?
+                %i.light redacted
```
Comment (eanders) [inline drivers/hud_apr/app/views/hud_apr/shared/_list.haml]: Is this not covered by `client.display_value(k, pii_policy: pii_policy, include_content_tag: true)`?  Feels like we should make sure it is if at all possible.

File: drivers/hud_apr/spec/controllers/hud_apr/apr/cells/search_queries_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                                          apr_id: report.id,
+                                          question_id: 'Question 5',
+                                          id: 'B2',
+                                          query_id: query.id,
+                                          table: '5a',
+                                        ))
+      end
+    end
+
+    context 'with unauthorized user' do
```
Comment (eanders) [inline drivers/hud_apr/spec/controllers/hud_apr/apr/cells/search_queries_controller_spec.rb]: Should we also check `can_view_all_hud_reports` and confirm they _can_ see it?

**Comment by ttoomey** [comment]
@eanders - thanks for the feedback, sorry this wasn't quite ready for review yet; it's still too raw

**Comment by eanders** [comment]
@ttoomey, no worries, I was just getting a jump on it.  I actually think it looks really close.

File: app/models/hud_reports/cell_detail_export_builder_base.rb (hunk 2 above)
Comment (ttoomey) [reply app/models/hud_reports/cell_detail_export_builder_base.rb]: good idea, but I might wait until we need it before adding indirection

File: drivers/hud_spm_report/app/models/hud_spm_report/document_exports/cell_detail_export.rb [hunk 6]
```diff
@@ -24,7 +24,7 @@ def perform
       end
 
       def download_title
-        "SPM Cell Detail – #{params.fetch('measure_id')} #{params.fetch('cell_id')}"
+        "SPM Cell Detail - #{params.fetch('measure_id')} #{params.fetch('cell_id')}"
```
Comment (ttoomey) [inline drivers/hud_spm_report/app/models/hud_spm_report/document_exports/cell_detail_export.rb]: the dash was causing and encoding error when sending mail

File: drivers/hud_spm_report/app/models/hud_spm_report/cell_detail_export_builder.rb [hunk 7]
```diff
@@ -1,41 +1,21 @@
...
+
 # frozen_string_literal: true
 
 module HudSpmReport
-  class CellDetailExportBuilder
-    require 'axlsx'
-
-    Result = Struct.new(:name, :filename, :data, keyword_init: true)
-
+  class CellDetailExportBuilder < ::HudReports::CellDetailExportBuilderBase
```
Comment (ttoomey) [inline drivers/hud_spm_report/app/models/hud_spm_report/cell_detail_export_builder.rb:10]: extracted to base class to share with spm

File: lib/hud_reports/route_concerns.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+# Shared route patterns for HUD reports across driver modules.
+module HudReports
+  module RouteConcerns
```
Comment (ttoomey) [inline lib/hud_reports/route_concerns.rb:11]: drys up repeated routing, only applied the the SPM and APR but could be used in the PIT, PATH, DQ

**Comment by ttoomey** [comment]
# AI Suggested QA Instructions

## What Changed
Cell drilldown views in APR, CAPER, CE APR, and DQ reports now include pagination and client search functionality. Excel exports for these reports are now backgrounded and delivered via email.

## Testing Focus

**Primary Testing** (Must verify):
- Pagination displays 100 records per page in APR/CAPER/CE APR/DQ cell drilldowns
- Client search filters results by name, ID, or Personal ID
- Excel exports trigger email delivery instead of direct download
- Search persists when navigating between pages

**Regression Check** (Quick verification):
- SPM report cell drilldown continues to work (already had these features)
- Non-owner users without "View All HUD Reports" permission cannot access others' report cells
- Cell data displays correctly with proper PII redaction based on project permissions

## Test Steps

### 1. Test APR Cell Drilldown Pagination (New Feature)

1. Navigate to any completed APR report
2. Click into a question result (e.g., "Question 5")
3. Click on a cell in the results table that has more than 100 records
4. **Verify**: Page displays "Showing 1-100 of [total]" with pagination controls
5. Click "Next" or page 2
6. **Verify**: Page displays records 101-200
7. **Verify**: Row numbers continue sequentially (101, 102, etc.)

### 2. Test Client Search in APR Cell Drilldown (New Feature)

1. From a cell drilldown view, locate the search box at the top
2. Enter a client's first name (e.g., "John")
3. Click "Search"
4. **Verify**: Results filter to show only matching clients
5. **Verify**: Count shows "Showing X of [total]" where X is filtered count
6. Click "Clear"
7. **Verify**: Full result set returns
8. Try searching by:
   - Last name
   - Personal ID
   - Client ID (numeric)
9. **Verify**: Each search type returns relevant results

### 3. Test Excel Export (Behavior Change)

1. From any APR cell drilldown view, click the "Download" button (Excel icon)
2. **Verify**: Modal appears: "Please wait, your request is being processed. You can continue to use the site and return for your download."
3. Wait 1-2 minutes, verify that modal shows "Processing complete. Your file is available for download." with Download link
5. **Verify**: Email arrives with download link
6. Click download link
7. **Verify**: Excel file downloads with cell detail data
8. **Verify**: PII columns are included/excluded based on configuration setting

### 4. Test Search Persistence Across Pages

1. In an APR cell drilldown with >100 results, perform a search that returns >100 results ("sa")
2. Click to page 2 of search results
3. **Verify**: Search term remains visible in search box
4. **Verify**: Results on page 2 are still filtered by search term
5. **Verify**: Pagination shows filtered count, not total

### 5. Test Other Report Types (Quick Check)

Repeat steps 1-4 for:
- **CAPER** report cell drilldown
- **CE APR** report cell drilldown  
- **DQ** (Data Quality) report cell drilldown

**Verify**: All show pagination and search functionality

### 6. SPM Report Regression Check

1. Navigate to a completed SPM report
2. Click into a measure result
3. Click on a cell with >100 records
4. **Verify**: Pagination still works (this already existed)
5. **Verify**: Search still works (this already existed)
6. **Verify**: Excel export triggers email (behavior changed from direct download)

**Note**: Unit tests cover edge cases like invalid cell IDs, expired search queries, and missing table parameters. Focus QA on the happy path and cross-report consistency.

## Known Limitations

- Search only works for models that have search capability implemented (AprClient). If search box doesn't appear, the model doesn't support search yet.
- HOPWA CAPER, HIC, PATH, and PIT reports were not updated in this change - they still use the old cell drilldown pattern without pagination or search.

File: app/models/hud_reports/report_instance.rb [hunk 9]
```diff
@@ -306,16 +306,6 @@ def generated_uploadable_version?
       question_names.include?('HDX Upload')
     end
 
-    # only allow alpha numeric
```
Comment (ttoomey) [inline app/models/hud_reports/report_instance.rb]: moved methods to generator, keep ReportInstance focused on persistence and relationships

File: app/models/hud_reports/report_cell.rb [hunk 10]
```diff
@@ -99,23 +99,28 @@ def completed?
     def write_detail(path:, generator:, question_name:)
```
Comment (ttoomey) [inline app/models/hud_reports/report_cell.rb:99]: This method appears to be dead code although perhaps devs use it for debugging? To remove:
```diff
diff --git a/app/models/hud_reports/report_cell.rb b/app/models/hud_reports/report_cell.rb
index 2cf7987cee..648b86a630 100644
--- a/app/models/hud_reports/report_cell.rb
+++ b/app/models/hud_reports/report_cell.rb
@@ -95,41 +95,6 @@ module HudReports
       status == 'Completed'
     end
 
-    # Render XLSX download to string, and save to path
-    def write_detail(path:, generator:, question_name:)
-      return unless path.present?
-
-      drilldown = generator.drilldown_context(
-        report: report_instance,
-        measure_id: question_name,
-        cell_id: cell_name,
-        table_id: question,
-      )
-
-      name = drilldown.name
-      headers = drilldown.headers
-      clients = drilldown.base_scope
-
-      template = generator.detail_template
-      xlsx_data = ApplicationController.render(
-        template: template,
-        formats: [:xlsx],
-        assigns: {
-          report: report_instance,
-          drilldown: drilldown,
-          # Legacy assigns for existing templates
-          question: drilldown.measure,
-          table: drilldown.table,
-          cell: drilldown.cell,
-          name: name,
-          headers: headers,
-          clients: clients,
-        },
-      )
-      cell_path = "#{path}#{question}-#{cell_name}.xlsx"
-      File.binwrite(cell_path, xlsx_data)
-    end
-
     # NOTE: sometimes warehouse_client isn't a client at all, but potentially an inventory record
     private def new_member(warehouse_client:, universe_client:)
       if universe_client.respond_to?(:first_name)
diff --git a/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2024/generator.rb b/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2024/generator.rb
index 6d6a366d7e..22cf7f7f20 100644
--- a/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2024/generator.rb
+++ b/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2024/generator.rb
@@ -8,8 +8,6 @@
 
 module  HudPit::Generators::Pit::Fy2024
   class Generator < ::HudReports::GeneratorBase
-    cattr_accessor :write_detail_path
-
     def self.fiscal_year
       'FY 2024'
     end
diff --git a/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2025/generator.rb b/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2025/generator.rb
index 15a83b5c30..2a0b54e39b 100644
--- a/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2025/generator.rb
+++ b/drivers/hud_pit/app/models/hud_pit/generators/pit/fy2025/generator.rb
@@ -8,8 +8,6 @@
 
 module  HudPit::Generators::Pit::Fy2025
   class Generator < ::HudReports::GeneratorBase
-    cattr_accessor :write_detail_path
-
     def self.fiscal_year
       'FY 2025'
     end
diff --git a/drivers/hud_pit/spec/models/datalab_pit_context.rb b/drivers/hud_pit/spec/models/datalab_pit_context.rb
index 01a2222490..221b1669a4 100644
--- a/drivers/hud_pit/spec/models/datalab_pit_context.rb
+++ b/drivers/hud_pit/spec/models/datalab_pit_context.rb
@@ -37,8 +37,6 @@ RSpec.shared_context 'datalab pit context', shared_context: :metadata do
       klass.title,
       build_for_questions: question_numbers,
     )
-    # Uncomment to get detail CSVs
-    # klass.write_detail_path = 'tmp/pit_'
     @generator = klass.new(report)
     @generator.run!
 
diff --git a/drivers/hud_pit/spec/models/hud_pit/generators/pit/fy2025/hud_pit_context.rb b/drivers/hud_pit/spec/models/hud_pit/generators/pit/fy2025/hud_pit_context.rb
index 694c4f6279..5dba10f70a 100644
--- a/drivers/hud_pit/spec/models/hud_pit/generators/pit/fy2025/hud_pit_context.rb
+++ b/drivers/hud_pit/spec/models/hud_pit/generators/pit/fy2025/hud_pit_context.rb
@@ -128,8 +128,6 @@ RSpec.shared_context 'HUD pit context', shared_context: :metadata do
       klass.title,
       build_for_questions: questions,
     )
-    # Uncomment to get detail CSVs
-    # klass.write_detail_path = 'tmp/pit_'
     generator = klass.new(report)
     generator.run!
 
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/generator.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/generator.rb
index 6ab2358300..f2a60fa5e2 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/generator.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/generator.rb
@@ -8,8 +8,6 @@
 
 module HudSpmReport::Generators::Fy2023
   class Generator < ::HudReports::GeneratorBase
-    cattr_accessor :write_detail_path
-
     def self.fiscal_year
       'FY 2023'
     end
@@ -65,10 +63,6 @@ module HudSpmReport::Generators::Fy2023
       ['enrollment.first_name', 'first_name', 'enrollment.last_name', 'last_name', 'dob', 'ssn']
     end
 
-    def self.detail_template
-      'hud_spm_report/cells/show'
-    end
-
     def self.uploadable_version?
       true
     end
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_base.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_base.rb
index b958852f51..8e290e5381 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_base.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_base.rb
@@ -58,13 +58,5 @@ module HudSpmReport::Generators::Fy2023
     private def generator_klass
       HudSpmReport::Generators::Fy2023::Generator
     end
-
-    private def write_detail(answer)
-      answer.write_detail(
-        path: generator_klass.write_detail_path,
-        generator: generator_klass,
-        question_name: self.class.question_number,
-      )
-    end
   end
 end
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_one.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_one.rb
index a8620b3fef..1dea6e764a 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_one.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2023/measure_one.rb
@@ -78,15 +78,12 @@ module HudSpmReport::Generators::Fy2023
       answer.update(summary: persons)
       # puts 'M1A B1'
       # puts cell_universe.map(&:universe_membership).map(&:client).map(&:personal_id)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :D1)
       answer.add_members(cell_universe)
       answer.update(summary: mean)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :G1)
       answer.add_members(cell_universe)
       answer.update(summary: median)
-      # write_detail(answer)
 
       create_universe(
         :m1a2,
@@ -137,7 +134,6 @@ module HudSpmReport::Generators::Fy2023
       answer.update(summary: persons)
       # puts 'M1B B1'
       # puts cell_universe.map(&:universe_membership).map(&:client).map(&:personal_id)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :D1)
       answer.add_members(cell_universe)
       answer.update(summary: mean)
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/generator.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/generator.rb
index f6d4faf78d..3bba3d58f5 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/generator.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/generator.rb
@@ -8,8 +8,6 @@
 
 module HudSpmReport::Generators::Fy2024
   class Generator < ::HudReports::GeneratorBase
-    cattr_accessor :write_detail_path
-
     def self.fiscal_year
       'FY 2024'
     end
@@ -65,10 +63,6 @@ module HudSpmReport::Generators::Fy2024
       ['enrollment.first_name', 'first_name', 'enrollment.last_name', 'last_name', 'dob', 'ssn']
     end
 
-    def self.detail_template
-      'hud_spm_report/cells/show'
-    end
-
     def self.uploadable_version?
       true
     end
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_base.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_base.rb
index d2b0189f6b..4082a6e365 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_base.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_base.rb
@@ -57,13 +57,5 @@ module HudSpmReport::Generators::Fy2024
     private def generator_klass
       HudSpmReport::Generators::Fy2024::Generator
     end
-
-    private def write_detail(answer)
-      answer.write_detail(
-        path: generator_klass.write_detail_path,
-        generator: generator_klass,
-        question_name: self.class.question_number,
-      )
-    end
   end
 end
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_one.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_one.rb
index e9b71c0aa3..23264645de 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_one.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2024/measure_one.rb
@@ -80,15 +80,12 @@ module HudSpmReport::Generators::Fy2024
       answer.update(summary: persons)
       # puts 'M1A B1'
       # puts cell_universe.map(&:universe_membership).map(&:client).map(&:personal_id)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :D1)
       answer.add_members(cell_universe)
       answer.update(summary: mean)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :G1)
       answer.add_members(cell_universe)
       answer.update(summary: median)
-      # write_detail(answer)
 
       create_universe(
         :m1a2,
@@ -139,7 +136,6 @@ module HudSpmReport::Generators::Fy2024
       answer.update(summary: persons)
       # puts 'M1B B1'
       # puts cell_universe.map(&:universe_membership).map(&:client).map(&:personal_id)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :D1)
       answer.add_members(cell_universe)
       answer.update(summary: mean)
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/generator.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/generator.rb
index d6788ed735..77ec8ea6ef 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/generator.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/generator.rb
@@ -8,8 +8,6 @@
 
 module HudSpmReport::Generators::Fy2026
   class Generator < ::HudReports::GeneratorBase
-    cattr_accessor :write_detail_path
-
     def self.fiscal_year
       'FY 2026'
     end
@@ -69,10 +67,6 @@ module HudSpmReport::Generators::Fy2026
       ['enrollment.first_name', 'first_name', 'enrollment.last_name', 'last_name', 'dob', 'ssn']
     end
 
-    def self.detail_template
-      'hud_spm_report/cells/show'
-    end
-
     def self.uploadable_version?
       true
     end
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_base.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_base.rb
index 4ec9771079..befccbd2ee 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_base.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_base.rb
@@ -63,13 +63,5 @@ module HudSpmReport::Generators::Fy2026
     private def generator_klass
       HudSpmReport::Generators::Fy2026::Generator
     end
-
-    private def write_detail(answer)
-      answer.write_detail(
-        path: generator_klass.write_detail_path,
-        generator: generator_klass,
-        question_name: self.class.question_number,
-      )
-    end
   end
 end
diff --git a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_one.rb b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_one.rb
index d24fc9be24..827fbd96fa 100644
--- a/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_one.rb
+++ b/drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2026/measure_one.rb
@@ -78,15 +78,12 @@ module HudSpmReport::Generators::Fy2026
       answer.update(summary: persons)
       # puts 'M1A B1'
       # puts cell_universe.map(&:universe_membership).map(&:client).map(&:personal_id)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :D2)
       answer.add_members(cell_universe)
       answer.update(summary: mean)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :G2)
       answer.add_members(cell_universe)
       answer.update(summary: median)
-      # write_detail(answer)
 
       create_universe(
         :m1a2,
@@ -136,7 +133,6 @@ module HudSpmReport::Generators::Fy2026
       answer.update(summary: persons)
       # puts 'M1B B1'
       # puts cell_universe.map(&:universe_membership).map(&:client).map(&:personal_id)
-      # write_detail(answer)
       answer = @report.answer(question: table_name, cell: :D2)
       answer.add_members(cell_universe)
       answer.update(summary: mean)
diff --git a/drivers/hud_spm_report/spec/models/datalab_testkit/spm_context.rb b/drivers/hud_spm_report/spec/models/datalab_testkit/spm_context.rb
index 3d6a34f029..65647b2b85 100644
--- a/drivers/hud_spm_report/spec/models/datalab_testkit/spm_context.rb
+++ b/drivers/hud_spm_report/spec/models/datalab_testkit/spm_context.rb
@@ -35,8 +35,6 @@ RSpec.shared_context 'datalab spm context', shared_context: :metadata do
       klass.title,
       build_for_questions: question_numbers,
     )
-    # Uncomment to get detail CSVs
-    klass.write_detail_path = 'tmp/spm_'
     @generator = klass.new(report)
     @generator.run!
 
```

File: drivers/hud_spm_report/spec/controllers/hud_spm_report/cells/search_queries_controller_spec.rb [hunk 11]
```diff
@@ -1,90 +0,0 @@
...
-# Copyright 2016 - 2025 Green River Data Analysis, LLC
-#
-# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
-###
-
-# frozen_string_literal: true
-
-require 'rails_helper'
-
-RSpec.describe HudSpmReport::Cells::SearchQueriesController, type: :request do
```
Comment (ttoomey) [inline drivers/hud_spm_report/spec/controllers/hud_spm_report/cells/search_queries_controller_spec.rb:11]: this is already covered by 
spec/models/grda_warehouse/client_search_query_spec.rb, spec/controllers/concerns/hud_reports/cell_drilldown_concern_spec.rb, and drivers/hud_spm_report/spec/controllers/hud_spm_report/cells/search_queries_controller_spec.rb

File: app/controllers/concerns/hud_reports/cell_drilldown_concern.rb [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+        type: export_class_name,
+        status: DocumentExportBehavior::PENDING_STATUS,
+        query_string: export_query_params.to_query,
+      )
+
+      export_job_class.perform_later(export_id: export.id)
+
+      flash[:notice] = 'Your cell-detail export is being generated. You will receive an email with a download link shortly.'
+      redirect_back(fallback_location: fallback_path)
+    end
```
Comment (eanders) [inline app/controllers/concerns/hud_reports/cell_drilldown_concern.rb]: Can we not use the usual Excel export process where we show a modal that eventually gets populated with a  download link?

File: app/models/hud_reports/drilldown_context.rb [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+        joins(hud_reports_universe_members: { report_cell: :report_instance }).
+        merge(::HudReports::ReportCell.for_table(table).for_cell(cell)).
+        merge(::HudReports::ReportInstance.where(id: report.id)).
+        distinct
+    end
+
+    def client_scope_for_measure
+      return generator.client_scope(measure) if generator.respond_to?(:client_scope)
+
+      generator.client_class(measure)
```
Comment (eanders) [inline app/models/hud_reports/drilldown_context.rb:98]: maybe eventually we should add a standard (required) method to the generator class so we can just call `generator.client_scope_for_measure(measure)` and leave the decision about where to pull that scope from to the generator.  This seems fine for now.

File: app/models/hud_reports/report_cell.rb (hunk 10 above)
Comment (eanders) [reply app/models/hud_reports/report_cell.rb:99]: I believe this is code to help troubleshoot discrepancies in the test kit.  It allows the tests to write out csv or excel files in the same shape as the test kit.  So it would only be used while working on the test kit.  @dtgreiner do you think this is still useful?

**Review by eanders — APPROVED** [review]
@ttoomey, this looks like a really great feature to me.  Thank you for taking the time to build it out so cleanly.

I wonder if in a future iteration we could add some tests to the end of the test kit to confirm the actual behavior of the drill downs using reports with clients in cells.  It makes me slightly nervous that all of these tests side-step actually ever generating a report and loading the detail pages.  We can do QA manually, but it would be nice to know we have automated testing to catch future regressions.

File: app/controllers/concerns/hud_reports/cell_drilldown_concern.rb (hunk 12 above)
Comment (ttoomey) [reply app/controllers/concerns/hud_reports/cell_drilldown_concern.rb]: Oh great catch, this code path is not actually used; sorry I forgot to clean that up 🤦 

The download button uses js to transform this to a post.

I'll remove this cruft. I also noticed there's some easy-to-extract shared code the the cell-details-header, I'll pull that out

File: app/models/hud_reports/drilldown_context.rb (hunk 13 above)
Comment (ttoomey) [reply app/models/hud_reports/drilldown_context.rb:98]: For sure, I'd love to remove the conditional logic

**Comment by ttoomey** [comment]
Changes since last review
-  Consolidated view logic into shared header and download button partials to ensure UI consistency, standardize path helpers.
- Cleanup `CellDrilldownConcern` and subclasses by removing redundant xls export methods

**Comment by ttoomey** [comment]
@eanders 

> I wonder if in a future iteration we could add some tests to the end of the test kit to confirm the actual behavior of the drill downs using reports with clients in cells. It makes me slightly nervous that all of these tests side-step actually ever generating a report and loading the detail pages. We can do QA manually, but it would be nice to know we have automated testing to catch future regressions.

Good idea, I added [an SPM integration test](https://github.com/greenriver/hmis-warehouse/pull/6170/files#diff-375db15bf2f68a05771e4b8fc802b73aa0ead292a7f776d97f4c5bf59232e883R12), does that address the concern?

**Comment by eanders** [comment]
> @eanders
> 
> > I wonder if in a future iteration we could add some tests to the end of the test kit to confirm the actual behavior of the drill downs using reports with clients in cells. It makes me slightly nervous that all of these tests side-step actually ever generating a report and loading the detail pages. We can do QA manually, but it would be nice to know we have automated testing to catch future regressions.
> 
> Good idea, I added [an SPM integration test](https://github.com/greenriver/hmis-warehouse/pull/6170/files#diff-375db15bf2f68a05771e4b8fc802b73aa0ead292a7f776d97f4c5bf59232e883R12), does that address the concern?

That's probably sufficient.  I wasn't suggesting anything additional as part of this work.

**Review by dtgreiner — APPROVED** [review]

**Comment by ttoomey** [comment]
Keeping this pr agains 196 so we can deploy to prod out-of-cycle. Will deploy to QA in another branch against 197
